fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/LogicalUniqueExpression.java

```diff
@@ -0,0 +1,123 @@
+import com.apple.foundationdb.record.query.plan.cascades.AliasMap;
+import com.apple.foundationdb.record.query.plan.cascades.CorrelationIdentifier;
+import com.apple.foundationdb.record.query.plan.cascades.ExpressionRef;
+import com.apple.foundationdb.record.query.plan.cascades.Quantifier;
+import com.apple.foundationdb.record.query.plan.cascades.TranslationMap;
+import com.apple.foundationdb.record.query.plan.cascades.explain.InternalPlannerGraphRewritable;
+import com.apple.foundationdb.record.query.plan.cascades.explain.NodeInfo;
+import com.apple.foundationdb.record.query.plan.cascades.explain.PlannerGraph;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A relational planner expression representing a stream of unique records. This expression has a single child which
+ * is also a {@link RelationalExpression}.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class LogicalUniqueExpression implements RelationalExpressionWithChildren, InternalPlannerGraphRewritable {
+    @Nonnull
+    private final Quantifier inner;
+
+    public LogicalUniqueExpression(@Nonnull ExpressionRef<RelationalExpression> innerRef) {
+        this(Quantifier.forEach(innerRef));
+    }
+
+    public LogicalUniqueExpression(@Nonnull Quantifier inner) {
+        this.inner = inner;
+    }
+
+    @Override
+    public int getRelationalChildCount() {
+        return 1;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends Quantifier> getQuantifiers() {
+        return ImmutableList.of(inner);
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return ImmutableSet.of();
+    }
+
+    @Nonnull
+    @Override
+    public LogicalUniqueExpression translateCorrelations(@Nonnull final TranslationMap translationMap, @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        return new LogicalUniqueExpression(Iterables.getOnlyElement(translatedQuantifiers));
+    }
+
+    @Nonnull
+    @Override
+    public Value getResultValue() {
+        return inner.getFlowedObjectValue();
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public boolean equalsWithoutChildren(@Nonnull final RelationalExpression otherExpression, @Nonnull final AliasMap equivalences) {
+        if (this == otherExpression) {
+            return true;
+        }
+        return getClass() == otherExpression.getClass();
+    }
+
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    @Override
+    public boolean equals(final Object other) {
+        return semanticEquals(other);
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return 31;
```

Comment:
Can you use a different prime? Half of our expressions use that number :-). Ideally we should use something like the `BASE_HASH` or maybe we can incorporate the hash of the class here since this hash code is never externalized and does not need to be stable, like plan hashes need to be.

Reply:
Good point! I am using a different prime number (`251`)